<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.7"/>
<title>pyEnsemblRest: ensembl._pyrest_server.RestServer Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/javascript">
  $(document).ready(initResizable);
  $(window).load(resizeHeight);
</script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { searchBox.OnSelectItem(0); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">pyEnsemblRest
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.7 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="annotated.html"><span>Class&#160;List</span></a></li>
      <li><a href="classes.html"><span>Class&#160;Index</span></a></li>
      <li><a href="inherits.html"><span>Class&#160;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&#160;Members</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
$(document).ready(function(){initNavTree('classensembl_1_1__pyrest__server_1_1RestServer.html','');});
</script>
<div id="doc-content">
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
<a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(0)"><span class="SelectionMark">&#160;</span>All</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(1)"><span class="SelectionMark">&#160;</span>Classes</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(2)"><span class="SelectionMark">&#160;</span>Namespaces</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(3)"><span class="SelectionMark">&#160;</span>Files</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(4)"><span class="SelectionMark">&#160;</span>Functions</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(5)"><span class="SelectionMark">&#160;</span>Variables</a><a class="SelectItem" href="javascript:void(0)" onclick="searchBox.OnSelectItem(6)"><span class="SelectionMark">&#160;</span>Properties</a></div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="#pri-methods">Private Member Functions</a> &#124;
<a href="#pri-static-attribs">Static Private Attributes</a> &#124;
<a href="classensembl_1_1__pyrest__server_1_1RestServer-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">ensembl._pyrest_server.RestServer Class Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html" title="RestServer is a class that knows how to communicate with the Ensembl REST servers. ">RestServer</a> is a class that knows how to communicate with the Ensembl REST servers.  
 <a href="classensembl_1_1__pyrest__server_1_1RestServer.html#details">More...</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a6a916102a59a5760878f14a202c43512"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a6a916102a59a5760878f14a202c43512">__init__</a></td></tr>
<tr class="memdesc:a6a916102a59a5760878f14a202c43512"><td class="mdescLeft">&#160;</td><td class="mdescRight">Constructor for the <a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html" title="RestServer is a class that knows how to communicate with the Ensembl REST servers. ">RestServer</a> class.  <a href="#a6a916102a59a5760878f14a202c43512">More...</a><br /></td></tr>
<tr class="separator:a6a916102a59a5760878f14a202c43512"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2b8617a853c671008adf1f37367752c1"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2b8617a853c671008adf1f37367752c1">getArchiveEntry</a></td></tr>
<tr class="memdesc:a2b8617a853c671008adf1f37367752c1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Uses the given identifier to return the archived sequence.  <a href="#a2b8617a853c671008adf1f37367752c1">More...</a><br /></td></tr>
<tr class="separator:a2b8617a853c671008adf1f37367752c1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3b3082403b5dc68881c98b852eb1531f"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3b3082403b5dc68881c98b852eb1531f">getAssemblyInfo</a></td></tr>
<tr class="memdesc:a3b3082403b5dc68881c98b852eb1531f"><td class="mdescLeft">&#160;</td><td class="mdescRight">List the currently available assemblies for a species.  <a href="#a3b3082403b5dc68881c98b852eb1531f">More...</a><br /></td></tr>
<tr class="separator:a3b3082403b5dc68881c98b852eb1531f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af2112ca2521e08e33febd15bffe16fed"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#af2112ca2521e08e33febd15bffe16fed">getAssemblyInfoRegion</a></td></tr>
<tr class="memdesc:af2112ca2521e08e33febd15bffe16fed"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns information about the specified toplevel sequence region for the given species.  <a href="#af2112ca2521e08e33febd15bffe16fed">More...</a><br /></td></tr>
<tr class="separator:af2112ca2521e08e33febd15bffe16fed"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a14dd5323a94b3e1e1e0ea806db79b907"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a14dd5323a94b3e1e1e0ea806db79b907">getGeneTreeById</a></td></tr>
<tr class="memdesc:a14dd5323a94b3e1e1e0ea806db79b907"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves a gene tree dump for a gene tree stable identifier.  <a href="#a14dd5323a94b3e1e1e0ea806db79b907">More...</a><br /></td></tr>
<tr class="separator:a14dd5323a94b3e1e1e0ea806db79b907"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad271a009012592b8bd1132b7e9cd666c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad271a009012592b8bd1132b7e9cd666c">getGeneTreeByMemberId</a></td></tr>
<tr class="memdesc:ad271a009012592b8bd1132b7e9cd666c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves a gene tree that contains the stable identifier.  <a href="#ad271a009012592b8bd1132b7e9cd666c">More...</a><br /></td></tr>
<tr class="separator:ad271a009012592b8bd1132b7e9cd666c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a45efa39906323c7eb5ef003fde876f2d"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a45efa39906323c7eb5ef003fde876f2d">getGeneTreeByMemberSymbol</a></td></tr>
<tr class="memdesc:a45efa39906323c7eb5ef003fde876f2d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves a gene tree containing the gene identified by a symbol.  <a href="#a45efa39906323c7eb5ef003fde876f2d">More...</a><br /></td></tr>
<tr class="separator:a45efa39906323c7eb5ef003fde876f2d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afdd757486abcff4a6838b1317b7022ba"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#afdd757486abcff4a6838b1317b7022ba">getHomologyByGeneSymbol</a></td></tr>
<tr class="memdesc:afdd757486abcff4a6838b1317b7022ba"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves homology information (orthologs) by symbol.  <a href="#afdd757486abcff4a6838b1317b7022ba">More...</a><br /></td></tr>
<tr class="separator:afdd757486abcff4a6838b1317b7022ba"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:affe1f93c1e1f810758d03d22e7b7e2bd"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#affe1f93c1e1f810758d03d22e7b7e2bd">getHomologyByGeneStableID</a></td></tr>
<tr class="memdesc:affe1f93c1e1f810758d03d22e7b7e2bd"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves homology information (orthologs) by Ensembl gene id.  <a href="#affe1f93c1e1f810758d03d22e7b7e2bd">More...</a><br /></td></tr>
<tr class="separator:affe1f93c1e1f810758d03d22e7b7e2bd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4961eae151292e29939a8a83bf5e6aae"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a4961eae151292e29939a8a83bf5e6aae">getGenomicAlignmentByRegion</a></td></tr>
<tr class="memdesc:a4961eae151292e29939a8a83bf5e6aae"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves genomic alignments as separate blocks based on a region and species.  <a href="#a4961eae151292e29939a8a83bf5e6aae">More...</a><br /></td></tr>
<tr class="separator:a4961eae151292e29939a8a83bf5e6aae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1e5af2358db2f18df52de359690ad446"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1e5af2358db2f18df52de359690ad446">ping</a></td></tr>
<tr class="memdesc:a1e5af2358db2f18df52de359690ad446"><td class="mdescLeft">&#160;</td><td class="mdescRight">Checks if the service is alive.  <a href="#a1e5af2358db2f18df52de359690ad446">More...</a><br /></td></tr>
<tr class="separator:a1e5af2358db2f18df52de359690ad446"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acbdb1f715bc69a5d1247ad8657a935d6"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#acbdb1f715bc69a5d1247ad8657a935d6">listSpecies</a></td></tr>
<tr class="memdesc:acbdb1f715bc69a5d1247ad8657a935d6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Lists all available species, their aliases, available adaptor groups and data release.  <a href="#acbdb1f715bc69a5d1247ad8657a935d6">More...</a><br /></td></tr>
<tr class="separator:acbdb1f715bc69a5d1247ad8657a935d6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a29bcf93ab6da343c01898a3249bfc1c7"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a29bcf93ab6da343c01898a3249bfc1c7">listComparaDatabases</a></td></tr>
<tr class="memdesc:a29bcf93ab6da343c01898a3249bfc1c7"><td class="mdescLeft">&#160;</td><td class="mdescRight">Lists all available comparative genomics databases and their data release.  <a href="#a29bcf93ab6da343c01898a3249bfc1c7">More...</a><br /></td></tr>
<tr class="separator:a29bcf93ab6da343c01898a3249bfc1c7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5758a4fe84a4025cf29af9388bf4c57c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a5758a4fe84a4025cf29af9388bf4c57c">ensembl_version</a></td></tr>
<tr class="memdesc:a5758a4fe84a4025cf29af9388bf4c57c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Shows the current version of the Ensembl API used by the REST server.  <a href="#a5758a4fe84a4025cf29af9388bf4c57c">More...</a><br /></td></tr>
<tr class="separator:a5758a4fe84a4025cf29af9388bf4c57c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a28582fe2c741d95c84538de986cce616"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a28582fe2c741d95c84538de986cce616">rest_version</a></td></tr>
<tr class="memdesc:a28582fe2c741d95c84538de986cce616"><td class="mdescLeft">&#160;</td><td class="mdescRight">Shows the current version of the Ensembl REST API.  <a href="#a28582fe2c741d95c84538de986cce616">More...</a><br /></td></tr>
<tr class="separator:a28582fe2c741d95c84538de986cce616"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a34b21d974cb859e1bd43f09519a991f2"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a34b21d974cb859e1bd43f09519a991f2">listAvailableReleases</a></td></tr>
<tr class="memdesc:a34b21d974cb859e1bd43f09519a991f2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Shows the data releases available on this REST server.  <a href="#a34b21d974cb859e1bd43f09519a991f2">More...</a><br /></td></tr>
<tr class="separator:a34b21d974cb859e1bd43f09519a991f2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2fc9686e263bf7148505d9a894dea8c8"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2fc9686e263bf7148505d9a894dea8c8">getAnalysisList</a></td></tr>
<tr class="memdesc:a2fc9686e263bf7148505d9a894dea8c8"><td class="mdescLeft">&#160;</td><td class="mdescRight">List the names of analyses involved in generating Ensembl data.  <a href="#a2fc9686e263bf7148505d9a894dea8c8">More...</a><br /></td></tr>
<tr class="separator:a2fc9686e263bf7148505d9a894dea8c8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af1c4afa4af4ba10f68b7e22e562d11d3"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#af1c4afa4af4ba10f68b7e22e562d11d3">getBiotypesBySpecies</a></td></tr>
<tr class="memdesc:af1c4afa4af4ba10f68b7e22e562d11d3"><td class="mdescLeft">&#160;</td><td class="mdescRight">List the functional classifications of gene models that Ensembl associates with a particular species.  <a href="#af1c4afa4af4ba10f68b7e22e562d11d3">More...</a><br /></td></tr>
<tr class="separator:af1c4afa4af4ba10f68b7e22e562d11d3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0ac9a1d0effb79a0a851bdc27e79c18f"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a0ac9a1d0effb79a0a851bdc27e79c18f">getExternalDatabasesBySpecies</a></td></tr>
<tr class="memdesc:a0ac9a1d0effb79a0a851bdc27e79c18f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Lists all available external sources for a species.  <a href="#a0ac9a1d0effb79a0a851bdc27e79c18f">More...</a><br /></td></tr>
<tr class="separator:a0ac9a1d0effb79a0a851bdc27e79c18f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acd6e1777ea75af866fa1a068bf9866d5"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#acd6e1777ea75af866fa1a068bf9866d5">getAllComparaMethods</a></td></tr>
<tr class="memdesc:acd6e1777ea75af866fa1a068bf9866d5"><td class="mdescLeft">&#160;</td><td class="mdescRight">List all compara analyses available (an analysis defines the type of comparative data).  <a href="#acd6e1777ea75af866fa1a068bf9866d5">More...</a><br /></td></tr>
<tr class="separator:acd6e1777ea75af866fa1a068bf9866d5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa5a793d563fcd3669660def0dc3c0020"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aa5a793d563fcd3669660def0dc3c0020">getSpeciesSetByComparaMethod</a></td></tr>
<tr class="memdesc:aa5a793d563fcd3669660def0dc3c0020"><td class="mdescLeft">&#160;</td><td class="mdescRight">List all collections of species analysed with the specified compara method.  <a href="#aa5a793d563fcd3669660def0dc3c0020">More...</a><br /></td></tr>
<tr class="separator:aa5a793d563fcd3669660def0dc3c0020"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a49d1054ef7f54d12f6e2689d0f41275d"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a49d1054ef7f54d12f6e2689d0f41275d">lookupIdentifier</a></td></tr>
<tr class="memdesc:a49d1054ef7f54d12f6e2689d0f41275d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find the species and database for a single identifier.  <a href="#a49d1054ef7f54d12f6e2689d0f41275d">More...</a><br /></td></tr>
<tr class="separator:a49d1054ef7f54d12f6e2689d0f41275d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab216ec4cd25019e26ed2c7f36dd653ab"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ab216ec4cd25019e26ed2c7f36dd653ab">lookupGeneSymbol</a></td></tr>
<tr class="memdesc:ab216ec4cd25019e26ed2c7f36dd653ab"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find the species and database for a symbol in a linked external database.  <a href="#ab216ec4cd25019e26ed2c7f36dd653ab">More...</a><br /></td></tr>
<tr class="separator:ab216ec4cd25019e26ed2c7f36dd653ab"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aff6b6b851a8290ccd31e9d4ad8faae90"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aff6b6b851a8290ccd31e9d4ad8faae90">mapCoordinatesBetweenAssemblies</a></td></tr>
<tr class="memdesc:aff6b6b851a8290ccd31e9d4ad8faae90"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert the co-ordinates of one assembly to another.  <a href="#aff6b6b851a8290ccd31e9d4ad8faae90">More...</a><br /></td></tr>
<tr class="separator:aff6b6b851a8290ccd31e9d4ad8faae90"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac03cfdf0f4311e40cc90b74ae235374e"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ac03cfdf0f4311e40cc90b74ae235374e">mapCDNACoordinatesToGenome</a></td></tr>
<tr class="memdesc:ac03cfdf0f4311e40cc90b74ae235374e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert from cDNA coordinates to genomic coordinates.  <a href="#ac03cfdf0f4311e40cc90b74ae235374e">More...</a><br /></td></tr>
<tr class="separator:ac03cfdf0f4311e40cc90b74ae235374e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aec0a48bef0d6b6382c5ea9e05bc2fe71"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aec0a48bef0d6b6382c5ea9e05bc2fe71">mapCDSCoordinatesToGenome</a></td></tr>
<tr class="memdesc:aec0a48bef0d6b6382c5ea9e05bc2fe71"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert from CDS coordinates to genomic coordinates.  <a href="#aec0a48bef0d6b6382c5ea9e05bc2fe71">More...</a><br /></td></tr>
<tr class="separator:aec0a48bef0d6b6382c5ea9e05bc2fe71"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2efad4a28b137e36fbbe3ca11ac211be"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2efad4a28b137e36fbbe3ca11ac211be">mapProteinCoordinatesToGenome</a></td></tr>
<tr class="memdesc:a2efad4a28b137e36fbbe3ca11ac211be"><td class="mdescLeft">&#160;</td><td class="mdescRight">Convert from protein (translation) coordinates to genomic coordinates.  <a href="#a2efad4a28b137e36fbbe3ca11ac211be">More...</a><br /></td></tr>
<tr class="separator:a2efad4a28b137e36fbbe3ca11ac211be"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3ac674608b80a55994ce1c2e97caced4"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3ac674608b80a55994ce1c2e97caced4">getOntologyByID</a></td></tr>
<tr class="memdesc:a3ac674608b80a55994ce1c2e97caced4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Search for an ontological term by its namespaced identifier.  <a href="#a3ac674608b80a55994ce1c2e97caced4">More...</a><br /></td></tr>
<tr class="separator:a3ac674608b80a55994ce1c2e97caced4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5fc174ef9d1ced6f27cea12098f2c49b"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a5fc174ef9d1ced6f27cea12098f2c49b">getOntologyByName</a></td></tr>
<tr class="memdesc:a5fc174ef9d1ced6f27cea12098f2c49b"><td class="mdescLeft">&#160;</td><td class="mdescRight">Search for a list of ontological terms by their name.  <a href="#a5fc174ef9d1ced6f27cea12098f2c49b">More...</a><br /></td></tr>
<tr class="separator:a5fc174ef9d1ced6f27cea12098f2c49b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2611a333829e3828df965d04753930d0"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2611a333829e3828df965d04753930d0">getAllAncestorsOfOntologyID</a></td></tr>
<tr class="memdesc:a2611a333829e3828df965d04753930d0"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reconstruct the entire ancestry of a term from is_a and part_of relationships.  <a href="#a2611a333829e3828df965d04753930d0">More...</a><br /></td></tr>
<tr class="separator:a2611a333829e3828df965d04753930d0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8819e7be3bf8cdaef64255386fec13d6"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a8819e7be3bf8cdaef64255386fec13d6">getAllDescendantsOfOntologyID</a></td></tr>
<tr class="memdesc:a8819e7be3bf8cdaef64255386fec13d6"><td class="mdescLeft">&#160;</td><td class="mdescRight">Find all the terms descended from a given term.  <a href="#a8819e7be3bf8cdaef64255386fec13d6">More...</a><br /></td></tr>
<tr class="separator:a8819e7be3bf8cdaef64255386fec13d6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad09a3a900fd17d8d2c2fc25d5ec3e15d"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad09a3a900fd17d8d2c2fc25d5ec3e15d">getOntologyAncestorChart</a></td></tr>
<tr class="memdesc:ad09a3a900fd17d8d2c2fc25d5ec3e15d"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reconstruct the entire ancestry of a term from is_a and part_of relationships.  <a href="#ad09a3a900fd17d8d2c2fc25d5ec3e15d">More...</a><br /></td></tr>
<tr class="separator:ad09a3a900fd17d8d2c2fc25d5ec3e15d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a06d8b6bd585b9160e2c0a94b2daae93a"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a06d8b6bd585b9160e2c0a94b2daae93a">getAllFeaturesOnFeatureID</a></td></tr>
<tr class="memdesc:a06d8b6bd585b9160e2c0a94b2daae93a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves features (e.g.  <a href="#a06d8b6bd585b9160e2c0a94b2daae93a">More...</a><br /></td></tr>
<tr class="separator:a06d8b6bd585b9160e2c0a94b2daae93a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a11163c3e6ab5a1e9829291860b6a732f"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a11163c3e6ab5a1e9829291860b6a732f">getAllFeaturesOnRegion</a></td></tr>
<tr class="memdesc:a11163c3e6ab5a1e9829291860b6a732f"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieves multiple types of features for a given region.  <a href="#a11163c3e6ab5a1e9829291860b6a732f">More...</a><br /></td></tr>
<tr class="separator:a11163c3e6ab5a1e9829291860b6a732f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4547db232f065eac4db0aee31dbf20b2"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a4547db232f065eac4db0aee31dbf20b2">getAllFeaturesOnTranslation</a></td></tr>
<tr class="memdesc:a4547db232f065eac4db0aee31dbf20b2"><td class="mdescLeft">&#160;</td><td class="mdescRight">Retrieve features related to a specific Translation as described by its stable ID (e.g.  <a href="#a4547db232f065eac4db0aee31dbf20b2">More...</a><br /></td></tr>
<tr class="separator:a4547db232f065eac4db0aee31dbf20b2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1b696fa0af6ba06a53957b90c6a0361e"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1b696fa0af6ba06a53957b90c6a0361e">getRegulatoryFeatureByID</a></td></tr>
<tr class="memdesc:a1b696fa0af6ba06a53957b90c6a0361e"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns a RegulatoryFeature given its stable ID (e.g.  <a href="#a1b696fa0af6ba06a53957b90c6a0361e">More...</a><br /></td></tr>
<tr class="separator:a1b696fa0af6ba06a53957b90c6a0361e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a47b3548a5e0176f45b04b18d0e1679ee"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a47b3548a5e0176f45b04b18d0e1679ee">getFeatureSequenceByID</a></td></tr>
<tr class="memdesc:a47b3548a5e0176f45b04b18d0e1679ee"><td class="mdescLeft">&#160;</td><td class="mdescRight">Request multiple types of sequence by stable identifier.  <a href="#a47b3548a5e0176f45b04b18d0e1679ee">More...</a><br /></td></tr>
<tr class="separator:a47b3548a5e0176f45b04b18d0e1679ee"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1831b25954738b0d954f2915257c26c5"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1831b25954738b0d954f2915257c26c5">getSequenceOfRegion</a></td></tr>
<tr class="memdesc:a1831b25954738b0d954f2915257c26c5"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns the genomic sequence of the specified region of the given species.  <a href="#a1831b25954738b0d954f2915257c26c5">More...</a><br /></td></tr>
<tr class="separator:a1831b25954738b0d954f2915257c26c5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a78d437dabc8beefd581da622bd24cefb"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a78d437dabc8beefd581da622bd24cefb">getTaxonomyEntryByID</a></td></tr>
<tr class="memdesc:a78d437dabc8beefd581da622bd24cefb"><td class="mdescLeft">&#160;</td><td class="mdescRight">Search for a taxonomic term by its identifier or name.  <a href="#a78d437dabc8beefd581da622bd24cefb">More...</a><br /></td></tr>
<tr class="separator:a78d437dabc8beefd581da622bd24cefb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a214e822ff1302249af6ba77177375d6a"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a214e822ff1302249af6ba77177375d6a">getTaxonomyClassificationByID</a></td></tr>
<tr class="memdesc:a214e822ff1302249af6ba77177375d6a"><td class="mdescLeft">&#160;</td><td class="mdescRight">Return the taxonomic classification of a taxon node.  <a href="#a214e822ff1302249af6ba77177375d6a">More...</a><br /></td></tr>
<tr class="separator:a214e822ff1302249af6ba77177375d6a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3cef7126e4eb7bf53e8ec0ee7a0c78fa"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3cef7126e4eb7bf53e8ec0ee7a0c78fa">getTaxonomyEntryByName</a></td></tr>
<tr class="memdesc:a3cef7126e4eb7bf53e8ec0ee7a0c78fa"><td class="mdescLeft">&#160;</td><td class="mdescRight">Search for a taxonomic id by a non-scientific name.  <a href="#a3cef7126e4eb7bf53e8ec0ee7a0c78fa">More...</a><br /></td></tr>
<tr class="separator:a3cef7126e4eb7bf53e8ec0ee7a0c78fa"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad9ef94b7e88cd5ddb3671efd0e538c08"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad9ef94b7e88cd5ddb3671efd0e538c08">getVariationByID</a></td></tr>
<tr class="memdesc:ad9ef94b7e88cd5ddb3671efd0e538c08"><td class="mdescLeft">&#160;</td><td class="mdescRight">Uses a variation identifier (e.g.  <a href="#ad9ef94b7e88cd5ddb3671efd0e538c08">More...</a><br /></td></tr>
<tr class="separator:ad9ef94b7e88cd5ddb3671efd0e538c08"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa11643f4c11ac605e58e1d8f2983486c"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aa11643f4c11ac605e58e1d8f2983486c">getVariantConsequencesByRegionAllele</a></td></tr>
<tr class="memdesc:aa11643f4c11ac605e58e1d8f2983486c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Fetch variant consequences.  <a href="#aa11643f4c11ac605e58e1d8f2983486c">More...</a><br /></td></tr>
<tr class="separator:aa11643f4c11ac605e58e1d8f2983486c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac085e7a91b67964be9fb4487711f50bb"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ac085e7a91b67964be9fb4487711f50bb">getVariantConsequencesByVariationID</a></td></tr>
<tr class="memdesc:ac085e7a91b67964be9fb4487711f50bb"><td class="mdescLeft">&#160;</td><td class="mdescRight">Fetch variant consequences based on a variation identifier.  <a href="#ac085e7a91b67964be9fb4487711f50bb">More...</a><br /></td></tr>
<tr class="separator:ac085e7a91b67964be9fb4487711f50bb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a116a371b6785f9ff418c084a0bcd1fa4"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a116a371b6785f9ff418c084a0bcd1fa4">getVariantConsequencesByHGVS</a></td></tr>
<tr class="memdesc:a116a371b6785f9ff418c084a0bcd1fa4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Fetch variant consequences based on a HGVS notation.  <a href="#a116a371b6785f9ff418c084a0bcd1fa4">More...</a><br /></td></tr>
<tr class="separator:a116a371b6785f9ff418c084a0bcd1fa4"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:a2bd4a8528904fd5e23ce4ddec3c90652"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2bd4a8528904fd5e23ce4ddec3c90652">server_url</a></td></tr>
<tr class="separator:a2bd4a8528904fd5e23ce4ddec3c90652"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a14663d4e51e10889625b7a917a9b5347"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a14663d4e51e10889625b7a917a9b5347">http</a></td></tr>
<tr class="separator:a14663d4e51e10889625b7a917a9b5347"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a41af4531182edc3f2597d8c7ecf2c33f"><td class="memItemLeft" align="right" valign="top">&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a41af4531182edc3f2597d8c7ecf2c33f">last_headers</a></td></tr>
<tr class="separator:a41af4531182edc3f2597d8c7ecf2c33f"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-methods"></a>
Private Member Functions</h2></td></tr>
<tr class="memitem:a911a9bd143963eea035e9188abd86e77"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77">__get_json_answer</a></td></tr>
<tr class="separator:a911a9bd143963eea035e9188abd86e77"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a071038622ab8bb772af3e343cc8288af"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af">__build_rest_answer</a></td></tr>
<tr class="separator:a071038622ab8bb772af3e343cc8288af"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pri-static-attribs"></a>
Static Private Attributes</h2></td></tr>
<tr class="memitem:a36aea40c33230383043e8291e5a9022e"><td class="memItemLeft" align="right" valign="top">dictionary&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a36aea40c33230383043e8291e5a9022e">__content_types</a></td></tr>
<tr class="separator:a36aea40c33230383043e8291e5a9022e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:adfef2141465bd19e31b86d7ebafc3524"><td class="memItemLeft" align="right" valign="top">dictionary&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html#adfef2141465bd19e31b86d7ebafc3524">__return_codes</a></td></tr>
<tr class="separator:adfef2141465bd19e31b86d7ebafc3524"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html" title="RestServer is a class that knows how to communicate with the Ensembl REST servers. ">RestServer</a> is a class that knows how to communicate with the Ensembl REST servers. </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00038">38</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>
</div><h2 class="groupheader">Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="a6a916102a59a5760878f14a202c43512"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.__init__ </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>server_url</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Constructor for the <a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html" title="RestServer is a class that knows how to communicate with the Ensembl REST servers. ">RestServer</a> class. </p>
<p>server_url is the URL of the server. </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00066">66</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

</div>
</div>
<h2 class="groupheader">Member Function Documentation</h2>
<a class="anchor" id="a071038622ab8bb772af3e343cc8288af"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.__build_rest_answer </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>new_object</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>allowed_formats</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>optional_params</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>accessor</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>url</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em> = <code>{}</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00102">102</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-0" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-0-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-0-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-0-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="249,5,442,46"/></map>
</div>
</p>

<p><div id="dynsection-1" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-1-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-1-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-1-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_icgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_icgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_icgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a071038622ab8bb772af3e343cc8288af_icgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2b8617a853c671008adf1f37367752c1" title="Uses the given identifier to return the archived sequence. " alt="" coords="291,5,485,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3b3082403b5dc68881c98b852eb1531f" title="List the currently available assemblies for a species. " alt="" coords="291,70,485,111"/><area shape="rect" id="node4" href="classensembl_1_1__pyrest__server_1_1RestServer.html#af2112ca2521e08e33febd15bffe16fed" title="Returns information about the specified toplevel sequence region for the given species. " alt="" coords="287,135,489,177"/><area shape="rect" id="node5" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a14dd5323a94b3e1e1e0ea806db79b907" title="Retrieves a gene tree dump for a gene tree stable identifier. " alt="" coords="291,201,485,242"/><area shape="rect" id="node6" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad271a009012592b8bd1132b7e9cd666c" title="Retrieves a gene tree that contains the stable identifier. " alt="" coords="281,266,495,307"/><area shape="rect" id="node7" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a45efa39906323c7eb5ef003fde876f2d" title="Retrieves a gene tree containing the gene identified by a symbol. " alt="" coords="265,331,511,373"/><area shape="rect" id="node8" href="classensembl_1_1__pyrest__server_1_1RestServer.html#afdd757486abcff4a6838b1317b7022ba" title="Retrieves homology information (orthologs) by symbol. " alt="" coords="272,397,504,438"/><area shape="rect" id="node9" href="classensembl_1_1__pyrest__server_1_1RestServer.html#affe1f93c1e1f810758d03d22e7b7e2bd" title="Retrieves homology information (orthologs) by Ensembl gene id. " alt="" coords="269,462,507,503"/><area shape="rect" id="node10" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a4961eae151292e29939a8a83bf5e6aae" title="Retrieves genomic alignments as separate blocks based on a region and species. " alt="" coords="263,527,513,569"/><area shape="rect" id="node11" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1e5af2358db2f18df52de359690ad446" title="Checks if the service is alive. " alt="" coords="291,593,485,634"/><area shape="rect" id="node12" href="classensembl_1_1__pyrest__server_1_1RestServer.html#acbdb1f715bc69a5d1247ad8657a935d6" title="Lists all available species, their aliases, available adaptor groups and data release. " alt="" coords="291,658,485,699"/><area shape="rect" id="node13" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a29bcf93ab6da343c01898a3249bfc1c7" title="Lists all available comparative genomics databases and their data release. " alt="" coords="289,723,487,765"/><area shape="rect" id="node14" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a5758a4fe84a4025cf29af9388bf4c57c" title="Shows the current version of the Ensembl API used by the REST server. " alt="" coords="291,789,485,830"/><area shape="rect" id="node15" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a28582fe2c741d95c84538de986cce616" title="Shows the current version of the Ensembl REST API. " alt="" coords="291,854,485,895"/><area shape="rect" id="node16" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a34b21d974cb859e1bd43f09519a991f2" title="Shows the data releases available on this REST server. " alt="" coords="291,919,485,961"/><area shape="rect" id="node17" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2fc9686e263bf7148505d9a894dea8c8" title="List the names of analyses involved in generating Ensembl data. " alt="" coords="291,985,485,1026"/><area shape="rect" id="node18" href="classensembl_1_1__pyrest__server_1_1RestServer.html#af1c4afa4af4ba10f68b7e22e562d11d3" title="List the functional classifications of gene models that Ensembl associates with a particular species..." alt="" coords="290,1050,486,1091"/><area shape="rect" id="node19" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a0ac9a1d0effb79a0a851bdc27e79c18f" title="Lists all available external sources for a species. " alt="" coords="260,1115,516,1157"/><area shape="rect" id="node20" href="classensembl_1_1__pyrest__server_1_1RestServer.html#acd6e1777ea75af866fa1a068bf9866d5" title="List all compara analyses available (an analysis defines the type of comparative data). " alt="" coords="288,1181,488,1222"/><area shape="rect" id="node21" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aa5a793d563fcd3669660def0dc3c0020" title="List all collections of species analysed with the specified compara method. " alt="" coords="257,1246,519,1287"/><area shape="rect" id="node22" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a49d1054ef7f54d12f6e2689d0f41275d" title="Find the species and database for a single identifier. " alt="" coords="291,1311,485,1353"/><area shape="rect" id="node23" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ab216ec4cd25019e26ed2c7f36dd653ab" title="Find the species and database for a symbol in a linked external database. " alt="" coords="291,1377,485,1418"/><area shape="rect" id="node24" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aff6b6b851a8290ccd31e9d4ad8faae90" title="Convert the co&#45;ordinates of one assembly to another. " alt="" coords="249,1442,527,1483"/><area shape="rect" id="node25" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ac03cfdf0f4311e40cc90b74ae235374e" title="Convert from cDNA coordinates to genomic coordinates. " alt="" coords="257,1507,519,1549"/><area shape="rect" id="node26" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aec0a48bef0d6b6382c5ea9e05bc2fe71" title="Convert from CDS coordinates to genomic coordinates. " alt="" coords="262,1573,514,1614"/><area shape="rect" id="node27" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2efad4a28b137e36fbbe3ca11ac211be" title="Convert from protein (translation) coordinates to genomic coordinates. " alt="" coords="256,1638,520,1679"/><area shape="rect" id="node28" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3ac674608b80a55994ce1c2e97caced4" title="Search for an ontological term by its namespaced identifier. " alt="" coords="291,1703,485,1745"/><area shape="rect" id="node29" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a5fc174ef9d1ced6f27cea12098f2c49b" title="Search for a list of ontological terms by their name. " alt="" coords="291,1769,485,1810"/><area shape="rect" id="node30" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2611a333829e3828df965d04753930d0" title="Reconstruct the entire ancestry of a term from is_a and part_of relationships. " alt="" coords="270,1834,506,1875"/><area shape="rect" id="node31" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a8819e7be3bf8cdaef64255386fec13d6" title="Find all the terms descended from a given term. " alt="" coords="261,1899,515,1941"/><area shape="rect" id="node32" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad09a3a900fd17d8d2c2fc25d5ec3e15d" title="Reconstruct the entire ancestry of a term from is_a and part_of relationships. " alt="" coords="279,1965,497,2006"/><area shape="rect" id="node33" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a06d8b6bd585b9160e2c0a94b2daae93a" title="Retrieves features (e.g. " alt="" coords="277,2030,499,2071"/><area shape="rect" id="node34" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a11163c3e6ab5a1e9829291860b6a732f" title="Retrieves multiple types of features for a given region. " alt="" coords="285,2095,491,2137"/><area shape="rect" id="node35" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a4547db232f065eac4db0aee31dbf20b2" title="Retrieve features related to a specific Translation as described by its stable ID (e..." alt="" coords="273,2161,503,2202"/><area shape="rect" id="node36" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1b696fa0af6ba06a53957b90c6a0361e" title="Returns a RegulatoryFeature given its stable ID (e.g. " alt="" coords="279,2226,497,2267"/><area shape="rect" id="node37" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a47b3548a5e0176f45b04b18d0e1679ee" title="Request multiple types of sequence by stable identifier. " alt="" coords="281,2291,495,2333"/><area shape="rect" id="node38" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1831b25954738b0d954f2915257c26c5" title="Returns the genomic sequence of the specified region of the given species. " alt="" coords="291,2357,485,2398"/><area shape="rect" id="node39" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a78d437dabc8beefd581da622bd24cefb" title="Search for a taxonomic term by its identifier or name. " alt="" coords="286,2422,490,2463"/><area shape="rect" id="node40" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a214e822ff1302249af6ba77177375d6a" title="Return the taxonomic classification of a taxon node. " alt="" coords="262,2487,514,2529"/><area shape="rect" id="node41" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3cef7126e4eb7bf53e8ec0ee7a0c78fa" title="Search for a taxonomic id by a non&#45;scientific name. " alt="" coords="275,2553,501,2594"/><area shape="rect" id="node42" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad9ef94b7e88cd5ddb3671efd0e538c08" title="Uses a variation identifier (e.g. " alt="" coords="291,2618,485,2659"/><area shape="rect" id="node43" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aa11643f4c11ac605e58e1d8f2983486c" title="Fetch variant consequences. " alt="" coords="275,2683,501,2739"/><area shape="rect" id="node44" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ac085e7a91b67964be9fb4487711f50bb" title="Fetch variant consequences based on a variation identifier. " alt="" coords="275,2763,501,2819"/><area shape="rect" id="node45" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a116a371b6785f9ff418c084a0bcd1fa4" title="Fetch variant consequences based on a HGVS notation. " alt="" coords="256,2842,520,2883"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a911a9bd143963eea035e9188abd86e77"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.__get_json_answer </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>url</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>content_type</em> = <code>None</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00072">72</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-2" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-2-trigger" src="closed.png" alt="+"/> Here is the caller graph for this function:</div>
<div id="dynsection-2-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-2-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a911a9bd143963eea035e9188abd86e77_icgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a911a9bd143963eea035e9188abd86e77_icgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a911a9bd143963eea035e9188abd86e77_icgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a911a9bd143963eea035e9188abd86e77_icgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,1410,442,1451"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2b8617a853c671008adf1f37367752c1" title="Uses the given identifier to return the archived sequence. " alt="" coords="534,5,727,46"/><area shape="rect" id="node4" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3b3082403b5dc68881c98b852eb1531f" title="List the currently available assemblies for a species. " alt="" coords="534,70,727,111"/><area shape="rect" id="node5" href="classensembl_1_1__pyrest__server_1_1RestServer.html#af2112ca2521e08e33febd15bffe16fed" title="Returns information about the specified toplevel sequence region for the given species. " alt="" coords="529,135,732,177"/><area shape="rect" id="node6" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a14dd5323a94b3e1e1e0ea806db79b907" title="Retrieves a gene tree dump for a gene tree stable identifier. " alt="" coords="534,201,727,242"/><area shape="rect" id="node7" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad271a009012592b8bd1132b7e9cd666c" title="Retrieves a gene tree that contains the stable identifier. " alt="" coords="524,266,737,307"/><area shape="rect" id="node8" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a45efa39906323c7eb5ef003fde876f2d" title="Retrieves a gene tree containing the gene identified by a symbol. " alt="" coords="507,331,754,373"/><area shape="rect" id="node9" href="classensembl_1_1__pyrest__server_1_1RestServer.html#afdd757486abcff4a6838b1317b7022ba" title="Retrieves homology information (orthologs) by symbol. " alt="" coords="515,397,747,438"/><area shape="rect" id="node10" href="classensembl_1_1__pyrest__server_1_1RestServer.html#affe1f93c1e1f810758d03d22e7b7e2bd" title="Retrieves homology information (orthologs) by Ensembl gene id. " alt="" coords="511,462,750,503"/><area shape="rect" id="node11" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a4961eae151292e29939a8a83bf5e6aae" title="Retrieves genomic alignments as separate blocks based on a region and species. " alt="" coords="506,527,755,569"/><area shape="rect" id="node12" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1e5af2358db2f18df52de359690ad446" title="Checks if the service is alive. " alt="" coords="534,593,727,634"/><area shape="rect" id="node13" href="classensembl_1_1__pyrest__server_1_1RestServer.html#acbdb1f715bc69a5d1247ad8657a935d6" title="Lists all available species, their aliases, available adaptor groups and data release. " alt="" coords="534,658,727,699"/><area shape="rect" id="node14" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a29bcf93ab6da343c01898a3249bfc1c7" title="Lists all available comparative genomics databases and their data release. " alt="" coords="532,723,729,765"/><area shape="rect" id="node15" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a5758a4fe84a4025cf29af9388bf4c57c" title="Shows the current version of the Ensembl API used by the REST server. " alt="" coords="534,789,727,830"/><area shape="rect" id="node16" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a28582fe2c741d95c84538de986cce616" title="Shows the current version of the Ensembl REST API. " alt="" coords="534,854,727,895"/><area shape="rect" id="node17" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a34b21d974cb859e1bd43f09519a991f2" title="Shows the data releases available on this REST server. " alt="" coords="534,919,727,961"/><area shape="rect" id="node18" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2fc9686e263bf7148505d9a894dea8c8" title="List the names of analyses involved in generating Ensembl data. " alt="" coords="534,985,727,1026"/><area shape="rect" id="node19" href="classensembl_1_1__pyrest__server_1_1RestServer.html#af1c4afa4af4ba10f68b7e22e562d11d3" title="List the functional classifications of gene models that Ensembl associates with a particular species..." alt="" coords="533,1050,729,1091"/><area shape="rect" id="node20" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a0ac9a1d0effb79a0a851bdc27e79c18f" title="Lists all available external sources for a species. " alt="" coords="503,1115,759,1157"/><area shape="rect" id="node21" href="classensembl_1_1__pyrest__server_1_1RestServer.html#acd6e1777ea75af866fa1a068bf9866d5" title="List all compara analyses available (an analysis defines the type of comparative data). " alt="" coords="531,1181,731,1222"/><area shape="rect" id="node22" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aa5a793d563fcd3669660def0dc3c0020" title="List all collections of species analysed with the specified compara method. " alt="" coords="500,1246,761,1287"/><area shape="rect" id="node23" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a49d1054ef7f54d12f6e2689d0f41275d" title="Find the species and database for a single identifier. " alt="" coords="534,1311,727,1353"/><area shape="rect" id="node24" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ab216ec4cd25019e26ed2c7f36dd653ab" title="Find the species and database for a symbol in a linked external database. " alt="" coords="534,1377,727,1418"/><area shape="rect" id="node25" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aff6b6b851a8290ccd31e9d4ad8faae90" title="Convert the co&#45;ordinates of one assembly to another. " alt="" coords="491,1442,770,1483"/><area shape="rect" id="node26" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ac03cfdf0f4311e40cc90b74ae235374e" title="Convert from cDNA coordinates to genomic coordinates. " alt="" coords="499,1507,762,1549"/><area shape="rect" id="node27" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aec0a48bef0d6b6382c5ea9e05bc2fe71" title="Convert from CDS coordinates to genomic coordinates. " alt="" coords="505,1573,757,1614"/><area shape="rect" id="node28" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2efad4a28b137e36fbbe3ca11ac211be" title="Convert from protein (translation) coordinates to genomic coordinates. " alt="" coords="499,1638,763,1679"/><area shape="rect" id="node29" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3ac674608b80a55994ce1c2e97caced4" title="Search for an ontological term by its namespaced identifier. " alt="" coords="534,1703,727,1745"/><area shape="rect" id="node30" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a5fc174ef9d1ced6f27cea12098f2c49b" title="Search for a list of ontological terms by their name. " alt="" coords="534,1769,727,1810"/><area shape="rect" id="node31" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a2611a333829e3828df965d04753930d0" title="Reconstruct the entire ancestry of a term from is_a and part_of relationships. " alt="" coords="513,1834,749,1875"/><area shape="rect" id="node32" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a8819e7be3bf8cdaef64255386fec13d6" title="Find all the terms descended from a given term. " alt="" coords="503,1899,758,1941"/><area shape="rect" id="node33" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad09a3a900fd17d8d2c2fc25d5ec3e15d" title="Reconstruct the entire ancestry of a term from is_a and part_of relationships. " alt="" coords="521,1965,740,2006"/><area shape="rect" id="node34" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a06d8b6bd585b9160e2c0a94b2daae93a" title="Retrieves features (e.g. " alt="" coords="519,2030,742,2071"/><area shape="rect" id="node35" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a11163c3e6ab5a1e9829291860b6a732f" title="Retrieves multiple types of features for a given region. " alt="" coords="527,2095,734,2137"/><area shape="rect" id="node36" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a4547db232f065eac4db0aee31dbf20b2" title="Retrieve features related to a specific Translation as described by its stable ID (e..." alt="" coords="516,2161,745,2202"/><area shape="rect" id="node37" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1b696fa0af6ba06a53957b90c6a0361e" title="Returns a RegulatoryFeature given its stable ID (e.g. " alt="" coords="521,2226,740,2267"/><area shape="rect" id="node38" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a47b3548a5e0176f45b04b18d0e1679ee" title="Request multiple types of sequence by stable identifier. " alt="" coords="523,2291,738,2333"/><area shape="rect" id="node39" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a1831b25954738b0d954f2915257c26c5" title="Returns the genomic sequence of the specified region of the given species. " alt="" coords="533,2357,728,2398"/><area shape="rect" id="node40" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a78d437dabc8beefd581da622bd24cefb" title="Search for a taxonomic term by its identifier or name. " alt="" coords="529,2422,733,2463"/><area shape="rect" id="node41" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a214e822ff1302249af6ba77177375d6a" title="Return the taxonomic classification of a taxon node. " alt="" coords="505,2487,757,2529"/><area shape="rect" id="node42" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a3cef7126e4eb7bf53e8ec0ee7a0c78fa" title="Search for a taxonomic id by a non&#45;scientific name. " alt="" coords="518,2553,743,2594"/><area shape="rect" id="node43" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ad9ef94b7e88cd5ddb3671efd0e538c08" title="Uses a variation identifier (e.g. " alt="" coords="534,2618,727,2659"/><area shape="rect" id="node44" href="classensembl_1_1__pyrest__server_1_1RestServer.html#aa11643f4c11ac605e58e1d8f2983486c" title="Fetch variant consequences. " alt="" coords="517,2683,744,2739"/><area shape="rect" id="node45" href="classensembl_1_1__pyrest__server_1_1RestServer.html#ac085e7a91b67964be9fb4487711f50bb" title="Fetch variant consequences based on a variation identifier. " alt="" coords="517,2763,744,2819"/><area shape="rect" id="node46" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a116a371b6785f9ff418c084a0bcd1fa4" title="Fetch variant consequences based on a HGVS notation. " alt="" coords="499,2842,763,2883"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a5758a4fe84a4025cf29af9388bf4c57c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.ensembl_version </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Shows the current version of the Ensembl API used by the REST server. </p>
<p>Return type: None Valid formats: json, xml HTTP endpoint: info/software </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00433">433</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-3" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-3-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-3-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-3-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a5758a4fe84a4025cf29af9388bf4c57c_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a5758a4fe84a4025cf29af9388bf4c57c_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a5758a4fe84a4025cf29af9388bf4c57c_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a5758a4fe84a4025cf29af9388bf4c57c_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2611a333829e3828df965d04753930d0"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAllAncestorsOfOntologyID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Reconstruct the entire ancestry of a term from is_a and part_of relationships. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1OntologyTerm.html">ensembl.info.OntologyTerm</a> Valid formats: json, xml, yaml HTTP endpoint: ontology/ancestors/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An ontology term identifier</li>
</ul>
<p>Optional parameters:</p><ul>
<li>ontology (String) Filter by ontology. Used to disambiguate terms which are shared between ontologies such as GO and EFO </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00746">746</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-4" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-4-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-4-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-4-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a2611a333829e3828df965d04753930d0_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a2611a333829e3828df965d04753930d0_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a2611a333829e3828df965d04753930d0_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a2611a333829e3828df965d04753930d0_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="291,5,485,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="534,5,727,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="acd6e1777ea75af866fa1a068bf9866d5"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAllComparaMethods </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>List all compara analyses available (an analysis defines the type of comparative data). </p>
<p>Return type: None Valid formats: json, json, yaml, xml HTTP endpoint: info/compara/methods</p>
<p>Optional parameters:</p><ul>
<li>compara (String) Name of the compara database to use. Multiple comparas may exist on a server when accessing Ensembl Genomes data.</li>
<li>class (String) The class of the method to query for. Regular expression patterns are supported. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00528">528</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-5" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-5-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-5-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-5-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_acd6e1777ea75af866fa1a068bf9866d5_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_acd6e1777ea75af866fa1a068bf9866d5_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_acd6e1777ea75af866fa1a068bf9866d5_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_acd6e1777ea75af866fa1a068bf9866d5_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="254,5,447,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="497,5,690,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a8819e7be3bf8cdaef64255386fec13d6"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAllDescendantsOfOntologyID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find all the terms descended from a given term. </p>
<p>By default searches are conducted within the namespace of the given identifier</p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1OntologyTerm.html">ensembl.info.OntologyTerm</a> Valid formats: json, xml HTTP endpoint: ontology/descendants/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An ontology term identifier</li>
</ul>
<p>Optional parameters:</p><ul>
<li>ontology (String) Filter by ontology. Used to disambiguate terms which are shared between ontologies such as GO and EFO</li>
<li>subset (String) Filter terms by the specified subset</li>
<li>closest_term (Boolean) If true return only the closest terms to the specified term</li>
<li>zero_distance (Boolean) Return terms with a distance of 0 </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00771">771</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-6" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-6-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-6-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-6-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a8819e7be3bf8cdaef64255386fec13d6_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a8819e7be3bf8cdaef64255386fec13d6_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a8819e7be3bf8cdaef64255386fec13d6_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a8819e7be3bf8cdaef64255386fec13d6_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="310,5,503,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="553,5,746,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a06d8b6bd585b9160e2c0a94b2daae93a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAllFeaturesOnFeatureID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves features (e.g. </p>
<p>genes, transcripts, variations etc.) that overlap a region defined by the given identifier.</p>
<p>Return type: <a class="el" href="namespaceensembl_1_1genome.html#a01899a318109093529836f503d161096" title="Wrapper arround the various types of features. ">ensembl.genome.feature_wrapper</a> Valid formats: json, xml, gff3, bed HTTP endpoint: overlap/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias.</li>
<li>object_type (String) Filter by feature type</li>
<li>db_type (String) Restrict the search to a database other than the default. Useful if you need to use a DB other than core</li>
<li>feature (Enum(gene, transcript, cds, exon, repeat, simple, misc, variation, somatic_variation, structural_variation, somatic_structural_variation, constrained, regulatory, segmentation, motif, chipseq, array_probe)) The type of feature to retrieve. Multiple values are accepted.</li>
<li>species_set (String) Filter by species set for retrieving constrained elements.</li>
<li>logic_name (String) Limit retrieval of genes, transcripts and exons by a given name of an analysis.</li>
<li>so_term (String) Sequence Ontology term to narrow down the possible variations returned.</li>
<li>misc_set (String) Miscellaneous set which groups together feature entries. Consult the DB or returned data sets to discover what is available.</li>
<li>biotype (String) The functional classification of the gene or transcript to fetch. Cannot be used in conjunction with logic_name when querying transcripts. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00825">825</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-7" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-7-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-7-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-7-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a06d8b6bd585b9160e2c0a94b2daae93a_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a06d8b6bd585b9160e2c0a94b2daae93a_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a06d8b6bd585b9160e2c0a94b2daae93a_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a06d8b6bd585b9160e2c0a94b2daae93a_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="278,5,471,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="521,5,714,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a11163c3e6ab5a1e9829291860b6a732f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAllFeaturesOnRegion </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves multiple types of features for a given region. </p>
<p>Return type: <a class="el" href="namespaceensembl_1_1genome.html#a01899a318109093529836f503d161096" title="Wrapper arround the various types of features. ">ensembl.genome.feature_wrapper</a> Valid formats: json, xml, gff3, bed HTTP endpoint: overlap/region/:species/:region</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias.</li>
<li>region (String) Query region. A maximum of 5Mb is allowed to be requested at any one time</li>
</ul>
<p>Optional parameters:</p><ul>
<li>db_type (String) Specify the database type to retrieve features from if not using the core database. We automatically choose the correct type of DB for variation, comparative and regulation features.</li>
<li>feature (Enum(gene, transcript, cds, exon, repeat, simple, misc, variation, somatic_variation, structural_variation, somatic_structural_variation, constrained, regulatory, segmentation, motif, chipseq, array_probe)) The type of feature to retrieve. Multiple values are accepted.</li>
<li>species_set (String) The species set name for retrieving constrained elements.</li>
<li>logic_name (String) Limit retrieval of genes, transcripts and exons by the name of analysis.</li>
<li>so_term (String) Sequence Ontology term to restrict the variations found. Its descendants are also included in the search.</li>
<li>cell_type (String) Cell type name in Ensembl's Regulatory Build, required for segmentation feature, optional for regulatory elements.</li>
<li>misc_set (String) Miscellaneous set which groups together feature entries. Consult the DB or returned data sets to discover what is available.</li>
<li>biotype (String) Functional classification of the gene or transcript to fetch. Cannot be used in conjunction with logic_name when querying transcripts.</li>
<li>trim_upstream (Boolean) Do not return features which overlap upstream end of the region.</li>
<li>trim_downstream (Boolean) Do not return features which overlap the downstream end of the region. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00864">864</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-8" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-8-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-8-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-8-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a11163c3e6ab5a1e9829291860b6a732f_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a11163c3e6ab5a1e9829291860b6a732f_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a11163c3e6ab5a1e9829291860b6a732f_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a11163c3e6ab5a1e9829291860b6a732f_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="262,5,455,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="505,5,698,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a4547db232f065eac4db0aee31dbf20b2"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAllFeaturesOnTranslation </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieve features related to a specific Translation as described by its stable ID (e.g. </p>
<p>domains, variations).</p>
<p>Return type: <a class="el" href="namespaceensembl_1_1genome.html#a01899a318109093529836f503d161096" title="Wrapper arround the various types of features. ">ensembl.genome.feature_wrapper</a> Valid formats: json, xml HTTP endpoint: overlap/translation/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias.</li>
<li>db_type (String) Restrict the search to a database other than the default. Useful if you need to use a DB other than core</li>
<li>feature (Enum(transcript_variation, protein_feature, residue_overlap, translation_exon, somatic_transcript_variation)) Specify the type of features requested for the translation.</li>
<li>type (String) Type of data to filter by. By default, all features are returned. Can specify a domain or consequence type.</li>
<li>so_term (String) Sequence Ontology term to restrict the variations found. Its descendants are also included in the search. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00891">891</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-9" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-9-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-9-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-9-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a4547db232f065eac4db0aee31dbf20b2_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a4547db232f065eac4db0aee31dbf20b2_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a4547db232f065eac4db0aee31dbf20b2_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a4547db232f065eac4db0aee31dbf20b2_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="283,5,477,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="526,5,719,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2fc9686e263bf7148505d9a894dea8c8"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAnalysisList </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>List the names of analyses involved in generating Ensembl data. </p>
<p>Return type: None Valid formats: json, xml HTTP endpoint: info/analysis/:species</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00475">475</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-10" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-10-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-10-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-10-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a2fc9686e263bf7148505d9a894dea8c8_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a2fc9686e263bf7148505d9a894dea8c8_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a2fc9686e263bf7148505d9a894dea8c8_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a2fc9686e263bf7148505d9a894dea8c8_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2b8617a853c671008adf1f37367752c1"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getArchiveEntry </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Uses the given identifier to return the archived sequence. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1ArchiveEntry.html">ensembl.info.ArchiveEntry</a> Valid formats: json, xml HTTP endpoint: archive/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00146">146</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-11" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-11-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-11-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-11-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a2b8617a853c671008adf1f37367752c1_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a2b8617a853c671008adf1f37367752c1_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a2b8617a853c671008adf1f37367752c1_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a2b8617a853c671008adf1f37367752c1_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a3b3082403b5dc68881c98b852eb1531f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAssemblyInfo </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>List the currently available assemblies for a species. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1Assembly.html">ensembl.info.Assembly</a> Valid formats: json, xml HTTP endpoint: info/assembly/:species</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
</ul>
<p>Optional parameters:</p><ul>
<li>bands (Boolean(0,1)) If set to 1, include karyotype band information. Only display if band information is available </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00165">165</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-12" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-12-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-12-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-12-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a3b3082403b5dc68881c98b852eb1531f_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a3b3082403b5dc68881c98b852eb1531f_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a3b3082403b5dc68881c98b852eb1531f_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a3b3082403b5dc68881c98b852eb1531f_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="af2112ca2521e08e33febd15bffe16fed"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getAssemblyInfoRegion </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region_name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Returns information about the specified toplevel sequence region for the given species. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1SeqRegion.html">ensembl.info.SeqRegion</a> Valid formats: json, xml HTTP endpoint: info/assembly/:species/:region_name</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>region_name (String) The (top level) sequence region name.</li>
</ul>
<p>Optional parameters:</p><ul>
<li>bands (Boolean(0,1)) If set to 1, include karyotype band information. Only display if band information is available </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00186">186</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-13" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-13-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-13-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-13-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_af2112ca2521e08e33febd15bffe16fed_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_af2112ca2521e08e33febd15bffe16fed_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_af2112ca2521e08e33febd15bffe16fed_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_af2112ca2521e08e33febd15bffe16fed_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="257,5,450,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="499,5,693,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="af1c4afa4af4ba10f68b7e22e562d11d3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getBiotypesBySpecies </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>List the functional classifications of gene models that Ensembl associates with a particular species. </p>
<p>Useful for restricting the type of genes/transcripts retrieved by other endpoints.</p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1Biotype.html">ensembl.info.Biotype</a> Valid formats: json, xml HTTP endpoint: info/biotypes/:species</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00491">491</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-14" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-14-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-14-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-14-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_af1c4afa4af4ba10f68b7e22e562d11d3_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_af1c4afa4af4ba10f68b7e22e562d11d3_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_af1c4afa4af4ba10f68b7e22e562d11d3_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_af1c4afa4af4ba10f68b7e22e562d11d3_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="251,5,445,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="494,5,687,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a0ac9a1d0effb79a0a851bdc27e79c18f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getExternalDatabasesBySpecies </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Lists all available external sources for a species. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1ExternalDatabase.html">ensembl.info.ExternalDatabase</a> Valid formats: json, xml HTTP endpoint: info/external_dbs/:species</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
</ul>
<p>Optional parameters:</p><ul>
<li>filter (String) Restrict external DB searches to a single source or pattern. SQL-LIKE patterns are supported. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00510">510</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-15" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-15-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-15-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-15-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a0ac9a1d0effb79a0a851bdc27e79c18f_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a0ac9a1d0effb79a0a851bdc27e79c18f_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a0ac9a1d0effb79a0a851bdc27e79c18f_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a0ac9a1d0effb79a0a851bdc27e79c18f_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="310,5,503,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="553,5,746,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a47b3548a5e0176f45b04b18d0e1679ee"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getFeatureSequenceByID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Request multiple types of sequence by stable identifier. </p>
<p>Return type: <a class="el" href="classensembl_1_1genome_1_1Sequence.html">ensembl.genome.Sequence</a> Valid formats: fasta, json, text, yaml HTTP endpoint: sequence/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>type (Enum(genomic,cds,cdna,protein)) Type of sequence. Defaults to genomic where applicable, i.e. not translations. cdna refers to the spliced transcript sequence with UTR; cds refers to the spliced transcript sequence without UTR.</li>
<li>species (String) Species name/alias</li>
<li>object_type (String) Filter by feature type</li>
<li>db_type (String) Restrict the search to a database other than the default. Useful if you need to use a DB other than core</li>
<li>format (Enum(fasta)) Format of the data</li>
<li>mask (Enum(hard,soft)) Request the sequence masked for repeat sequences. Hard will mask all repeats as N's and soft will mask repeats as lowercased characters. Only available when using genomic sequence type.</li>
<li>mask_feature (Boolean) Mask features on the sequence. If sequence is genomic, mask introns. If sequence is cDNA, mask UTRs. Incompatible with the 'mask' option</li>
<li>expand_5prime (Int) Expand the sequence upstream of the sequence by this many basepairs. Only available when using genomic sequence type.</li>
<li>expand_3prime (Int) Expand the sequence downstream of the sequence by this many basepairs. Only available when using genomic sequence type.</li>
<li>multiple_sequences (Boolean) Allow the service to return more than 1 sequence per identifier. This is useful when querying for a gene but using a type such as protein. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00946">946</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-16" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-16-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-16-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-16-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a47b3548a5e0176f45b04b18d0e1679ee_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a47b3548a5e0176f45b04b18d0e1679ee_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a47b3548a5e0176f45b04b18d0e1679ee_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a47b3548a5e0176f45b04b18d0e1679ee_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="270,5,463,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="513,5,706,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a14dd5323a94b3e1e1e0ea806db79b907"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getGeneTreeById </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves a gene tree dump for a gene tree stable identifier. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1GeneTree.html" title="Global object for gene-trees. ">ensembl.compara.GeneTree</a> Valid formats: phyloxml, nh, json HTTP endpoint: genetree/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl genetree ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>compara (String) Name of the compara database to use. Multiple comparas can exist on a server if you are accessing Ensembl Genomes data</li>
<li>aligned (Boolean) Return the aligned string if true. Otherwise, return the original sequence (no insertions)</li>
<li>sequence (Enum(none, cdna, protein)) The type of sequence to bring back. Setting it to none results in no sequence being returned</li>
<li>nh_format (Enum(full, display_label_composite, simple, species, species_short_name, ncbi_taxon, ncbi_name, njtree, phylip)) The format of a NH (New Hampshire) request. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00211">211</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-17" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-17-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-17-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-17-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a14dd5323a94b3e1e1e0ea806db79b907_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a14dd5323a94b3e1e1e0ea806db79b907_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a14dd5323a94b3e1e1e0ea806db79b907_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a14dd5323a94b3e1e1e0ea806db79b907_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ad271a009012592b8bd1132b7e9cd666c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getGeneTreeByMemberId </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves a gene tree that contains the stable identifier. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1GeneTree.html" title="Global object for gene-trees. ">ensembl.compara.GeneTree</a> Valid formats: phyloxml, nh, json HTTP endpoint: genetree/member/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>db_type (String) Restrict the search to a database other than the default. Useful if you need to use a DB other than core</li>
<li>object_type (String) Filter by feature type</li>
<li>compara (String) Name of the compara database to use. Multiple comparas can exist on a server if you are accessing Ensembl Genomes data</li>
<li>aligned (Boolean) Return the aligned string if true. Otherwise, return the original sequence (no insertions)</li>
<li>sequence (Enum(none, cdna, protein)) The type of sequence to bring back. Setting it to none results in no sequence being returned</li>
<li>nh_format (Enum(full, display_label_composite, simple, species, species_short_name, ncbi_taxon, ncbi_name, njtree, phylip)) The format of a NH (New Hampshire) request. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00242">242</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-18" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-18-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-18-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-18-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_ad271a009012592b8bd1132b7e9cd666c_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_ad271a009012592b8bd1132b7e9cd666c_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_ad271a009012592b8bd1132b7e9cd666c_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_ad271a009012592b8bd1132b7e9cd666c_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="267,5,461,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="510,5,703,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a45efa39906323c7eb5ef003fde876f2d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getGeneTreeByMemberSymbol </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>symbol</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves a gene tree containing the gene identified by a symbol. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1GeneTree.html" title="Global object for gene-trees. ">ensembl.compara.GeneTree</a> Valid formats: phyloxml, nh, json HTTP endpoint: genetree/member/symbol/:species/:symbol</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>symbol (String) Symbol or display name of a gene</li>
</ul>
<p>Optional parameters:</p><ul>
<li>db_type (String) Restrict the search to a database other than the default. Useful if you need to use a DB other than core</li>
<li>external_db (String) Filter by external database</li>
<li>object_type (String) Filter by feature type</li>
<li>compara (String) Name of the compara database to use. Multiple comparas can exist on a server if you are accessing Ensembl Genomes data</li>
<li>aligned (Boolean) Return the aligned string if true. Otherwise, return the original sequence (no insertions)</li>
<li>sequence (Enum(none, cdna, protein)) The type of sequence to bring back. Setting it to none results in no sequence being returned</li>
<li>nh_format (Enum(full, display_label_composite, simple, species, species_short_name, ncbi_taxon, ncbi_name, njtree, phylip)) The format of a NH (New Hampshire) request. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00275">275</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-19" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-19-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-19-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-19-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a45efa39906323c7eb5ef003fde876f2d_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a45efa39906323c7eb5ef003fde876f2d_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a45efa39906323c7eb5ef003fde876f2d_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a45efa39906323c7eb5ef003fde876f2d_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="302,5,495,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="545,5,738,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a4961eae151292e29939a8a83bf5e6aae"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getGenomicAlignmentByRegion </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves genomic alignments as separate blocks based on a region and species. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1GenomicAlignment.html">ensembl.compara.GenomicAlignment</a> Valid formats: json, xml, phyloxml HTTP endpoint: alignment/region/:species/:region</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>region (String) Query region. A maximum of 10Mb is allowed to be requested at any one time</li>
</ul>
<p>Optional parameters:</p><ul>
<li>compara (String) Name of the compara database to use. Multiple comparas can exist on a server if you are accessing Ensembl Genomes data</li>
<li>aligned (Boolean) Return the aligned string if true. Otherwise, return the original sequence (no insertions)</li>
<li>mask (Enum(hard,soft)) Request the sequence masked for repeat sequences. Hard will mask all repeats as N's and soft will mask repeats as lowercased characters.</li>
<li>species_set_group (String) The species set group name of the multiple alignment. Should not be used with the species_set parameter. Use <a href="/documentation/info/compara_species_sets">/info/compara/species_sets/:method</a> with one of the methods listed above to obtain a valid list of group names.</li>
<li>species_set (String) The set of species used to define the pairwise alignment (multiple values). Should not be used with the species_set_group parameter. Use <a href="/documentation/info/compara_species_sets">/info/compara/species_sets/:method</a> with one of the methods listed above to obtain a valid list of species sets. Any valid alias may be used.</li>
<li>method (Enum(EPO, EPO_LOW_COVERAGE, PECAN, LASTZ_NET, BLASTZ_NET, TRANSLATED_BLAT_NET)) The alignment method</li>
<li>display_species_set (String) Subset of species in the alignment to be displayed (multiple values). All the species in the alignment will be displayed if this is not set. Any valid alias may be used.</li>
<li>compact (Boolean) Applicable to EPO_LOW_COVERAGE alignments. If true, concatenate the low coverage species sequences together to create a single sequence. Otherwise, separates out all sequences. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00378">378</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-20" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-20-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-20-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-20-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a4961eae151292e29939a8a83bf5e6aae_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a4961eae151292e29939a8a83bf5e6aae_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a4961eae151292e29939a8a83bf5e6aae_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a4961eae151292e29939a8a83bf5e6aae_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="305,5,498,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="547,5,741,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="affe1f93c1e1f810758d03d22e7b7e2bd"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getHomologyByGeneStableID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves homology information (orthologs) by Ensembl gene id. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1HomologyGroup.html" title="Group of multiple homology-pairs. ">ensembl.compara.HomologyGroup</a> Valid formats: json, xml, orthoxml HTTP endpoint: homology/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>compara (String) Name of the compara database to use. Multiple comparas can exist on a server if you are accessing Ensembl Genomes data</li>
<li>format (Enum(full, condensed)) Layout of the response</li>
<li>type (Enum(orthologues, paralogues, projections, all)) The type of homology to return from this call. Projections are orthology calls defined between alternative assemblies and the genes shared between them. Useful if you need only one type of homology back from the service</li>
<li>target_species (String) Filter by species. Supports all species aliases</li>
<li>target_taxon (Integer) Filter by taxon</li>
<li>aligned (Boolean) Return the aligned string if true. Otherwise, return the original sequence (no insertions)</li>
<li>sequence (Enum(none, cdna, protein)) The type of sequence to bring back. Setting it to none results in no sequence being returned </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00343">343</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-21" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-21-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-21-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-21-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_affe1f93c1e1f810758d03d22e7b7e2bd_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_affe1f93c1e1f810758d03d22e7b7e2bd_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_affe1f93c1e1f810758d03d22e7b7e2bd_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_affe1f93c1e1f810758d03d22e7b7e2bd_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="294,5,487,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="537,5,730,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="afdd757486abcff4a6838b1317b7022ba"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getHomologyByGeneSymbol </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>symbol</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Retrieves homology information (orthologs) by symbol. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1HomologyGroup.html" title="Group of multiple homology-pairs. ">ensembl.compara.HomologyGroup</a> Valid formats: json, xml, orthoxml HTTP endpoint: homology/symbol/:species/:symbol</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>symbol (String) Symbol or display name of a gene</li>
</ul>
<p>Optional parameters:</p><ul>
<li>compara (String) Name of the compara database to use. Multiple comparas can exist on a server if you are accessing Ensembl Genomes data</li>
<li>external_db (String) Filter by external database</li>
<li>format (Enum(full,condensed)) Layout of the response</li>
<li>type (Enum(orthologues, paralogues, projections, all)) The type of homology to return from this call. Projections are orthology calls defined between alternative assemblies and the genes shared between them. Useful if you need only one type of homology back from the service</li>
<li>target_species (String) Filter by species. Supports all species aliases</li>
<li>target_taxon (Integer) Filter by taxon</li>
<li>aligned (Boolean) Return the aligned string if true. Otherwise, return the original sequence (no insertions)</li>
<li>sequence (Enum(none, cdna, protein)) The type of sequence to bring back. Setting it to none results in no sequence being returned </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00310">310</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-22" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-22-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-22-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-22-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_afdd757486abcff4a6838b1317b7022ba_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_afdd757486abcff4a6838b1317b7022ba_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_afdd757486abcff4a6838b1317b7022ba_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_afdd757486abcff4a6838b1317b7022ba_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="286,5,479,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="529,5,722,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ad09a3a900fd17d8d2c2fc25d5ec3e15d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getOntologyAncestorChart </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Reconstruct the entire ancestry of a term from is_a and part_of relationships. </p>
<p>Return type: Dictionary of String -&gt; <a class="el" href="classensembl_1_1info_1_1OntologyEntry.html">ensembl.info.OntologyEntry</a> Valid formats: json, xml HTTP endpoint: ontology/ancestors/chart/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An ontology term identifier</li>
</ul>
<p>Optional parameters:</p><ul>
<li>ontology (String) Filter by ontology. Used to disambiguate terms which are shared between ontologies such as GO and EFO </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00790">790</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-23" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-23-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-23-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-23-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_ad09a3a900fd17d8d2c2fc25d5ec3e15d_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_ad09a3a900fd17d8d2c2fc25d5ec3e15d_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_ad09a3a900fd17d8d2c2fc25d5ec3e15d_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_ad09a3a900fd17d8d2c2fc25d5ec3e15d_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="273,5,466,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="515,5,709,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a3ac674608b80a55994ce1c2e97caced4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getOntologyByID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Search for an ontological term by its namespaced identifier. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1OntologyTerm.html">ensembl.info.OntologyTerm</a> Valid formats: json, xml, yaml HTTP endpoint: ontology/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An ontology term identifier</li>
</ul>
<p>Optional parameters:</p><ul>
<li>relation (String) The types of relationships to include in the output. Fetches all relations by default</li>
<li>simple (Boolean) If set the API will avoid the fetching of parent and child terms </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00704">704</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-24" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-24-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-24-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-24-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a3ac674608b80a55994ce1c2e97caced4_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a3ac674608b80a55994ce1c2e97caced4_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a3ac674608b80a55994ce1c2e97caced4_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a3ac674608b80a55994ce1c2e97caced4_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a5fc174ef9d1ced6f27cea12098f2c49b"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getOntologyByName </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Search for a list of ontological terms by their name. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1OntologyTerm.html">ensembl.info.OntologyTerm</a> Valid formats: json, xml, yaml HTTP endpoint: ontology/name/:name</p>
<p>Required parameters:</p><ul>
<li>name (String) An ontology name. SQL wildcards are supported</li>
</ul>
<p>Optional parameters:</p><ul>
<li>ontology (String) Filter by ontology. Used to disambiguate terms which are shared between ontologies such as GO and EFO</li>
<li>relation (String) The types of relationships to include in the output. Fetches all relations by default</li>
<li>simple (Boolean) If set the API will avoid the fetching of parent and child terms </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00727">727</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-25" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-25-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-25-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-25-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a5fc174ef9d1ced6f27cea12098f2c49b_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a5fc174ef9d1ced6f27cea12098f2c49b_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a5fc174ef9d1ced6f27cea12098f2c49b_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a5fc174ef9d1ced6f27cea12098f2c49b_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1b696fa0af6ba06a53957b90c6a0361e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getRegulatoryFeatureByID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Returns a RegulatoryFeature given its stable ID (e.g. </p>
<p>ENSR00001348195)</p>
<p>Return type: ensembl.funcgen.RegulatoryFeature Valid formats: json, xml HTTP endpoint: regulatory/:species/:id</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>id (String) RegulatoryFeature stable ID </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00909">909</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-26" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-26-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-26-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-26-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a1b696fa0af6ba06a53957b90c6a0361e_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a1b696fa0af6ba06a53957b90c6a0361e_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a1b696fa0af6ba06a53957b90c6a0361e_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a1b696fa0af6ba06a53957b90c6a0361e_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="273,5,466,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="515,5,709,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1831b25954738b0d954f2915257c26c5"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getSequenceOfRegion </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Returns the genomic sequence of the specified region of the given species. </p>
<p>Return type: <a class="el" href="classensembl_1_1genome_1_1Sequence.html">ensembl.genome.Sequence</a> Valid formats: fasta, json, text, yaml HTTP endpoint: sequence/region/:species/:region</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>region (String) Query region. A maximum of 10Mb is allowed to be requested at any one time</li>
</ul>
<p>Optional parameters:</p><ul>
<li>format (Enum(fasta)) Format of the data.</li>
<li>mask (Enum(hard,soft)) Request the sequence masked for repeat sequences. Hard will mask all repeats as N's and soft will mask repeats as lower cased characters. Only available when using genomic sequence type.</li>
<li>mask_feature (Boolean) Mask features on the sequence. If sequence is genomic, mask introns. If sequence is cDNA, mask UTRs. Incompatible with the 'mask' option</li>
<li>expand_5prime (Int) Expand the sequence upstream of the sequence by this many basepairs. Only available when using genomic sequence type.</li>
<li>expand_3prime (Int) Expand the sequence downstream of the sequence by this many basepairs. Only available when using genomic sequence type.</li>
<li>coord_system (String) Filter by coordinate system name</li>
<li>coord_system_version (String) Filter by coordinate system version </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00979">979</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-27" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-27-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-27-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-27-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a1831b25954738b0d954f2915257c26c5_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a1831b25954738b0d954f2915257c26c5_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a1831b25954738b0d954f2915257c26c5_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a1831b25954738b0d954f2915257c26c5_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa5a793d563fcd3669660def0dc3c0020"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getSpeciesSetByComparaMethod </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>method</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>List all collections of species analysed with the specified compara method. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1MethodLinkSpeciesSet.html">ensembl.compara.MethodLinkSpeciesSet</a> Valid formats: json, yaml, xml HTTP endpoint: info/compara/species_sets/:method</p>
<p>Required parameters:</p><ul>
<li>method (String) Filter by compara method. Use one the methods returned by <a href="/documentation/info/compara_methods">/info/compara/methods</a> endpoint.</li>
</ul>
<p>Optional parameters:</p><ul>
<li>compara (String) Name of the compara database to use. Multiple comparas may exist on a server when accessing Ensembl Genomes data. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00547">547</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-28" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-28-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-28-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-28-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_aa5a793d563fcd3669660def0dc3c0020_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_aa5a793d563fcd3669660def0dc3c0020_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_aa5a793d563fcd3669660def0dc3c0020_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_aa5a793d563fcd3669660def0dc3c0020_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="315,5,509,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="558,5,751,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a214e822ff1302249af6ba77177375d6a"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getTaxonomyClassificationByID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Return the taxonomic classification of a taxon node. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1NCBITaxon.html" title="A node in the NCBI taxonomy. ">ensembl.compara.NCBITaxon</a> Valid formats: json, xml, yaml HTTP endpoint: taxonomy/classification/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) A taxon identifier. Can be a NCBI taxon id or a name </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l01014">1014</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-29" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-29-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-29-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-29-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a214e822ff1302249af6ba77177375d6a_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a214e822ff1302249af6ba77177375d6a_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a214e822ff1302249af6ba77177375d6a_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a214e822ff1302249af6ba77177375d6a_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="307,5,501,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="550,5,743,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a78d437dabc8beefd581da622bd24cefb"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getTaxonomyEntryByID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Search for a taxonomic term by its identifier or name. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1NCBITaxon.html" title="A node in the NCBI taxonomy. ">ensembl.compara.NCBITaxon</a> Valid formats: json, xml, yaml HTTP endpoint: taxonomy/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) A taxon identifier. Can be a NCBI taxon id or a name</li>
</ul>
<p>Optional parameters:</p><ul>
<li>simple (Boolean) If set the API will avoid the fetching of parent and child terms </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00998">998</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-30" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-30-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-30-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-30-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a78d437dabc8beefd581da622bd24cefb_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a78d437dabc8beefd581da622bd24cefb_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a78d437dabc8beefd581da622bd24cefb_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a78d437dabc8beefd581da622bd24cefb_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="259,5,453,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="502,5,695,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a3cef7126e4eb7bf53e8ec0ee7a0c78fa"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getTaxonomyEntryByName </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>name</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Search for a taxonomic id by a non-scientific name. </p>
<p>Return type: <a class="el" href="classensembl_1_1compara_1_1NCBITaxon.html" title="A node in the NCBI taxonomy. ">ensembl.compara.NCBITaxon</a> Valid formats: json, xml, yaml HTTP endpoint: taxonomy/name/:name</p>
<p>Required parameters:</p><ul>
<li>name (String) A non-scientific species name. Can include SQL wildcards </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l01030">1030</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-31" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-31-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-31-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-31-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a3cef7126e4eb7bf53e8ec0ee7a0c78fa_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a3cef7126e4eb7bf53e8ec0ee7a0c78fa_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a3cef7126e4eb7bf53e8ec0ee7a0c78fa_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a3cef7126e4eb7bf53e8ec0ee7a0c78fa_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="281,5,474,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="523,5,717,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a116a371b6785f9ff418c084a0bcd1fa4"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getVariantConsequencesByHGVS </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>hgvs_notation</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Fetch variant consequences based on a HGVS notation. </p>
<p>Return type: <a class="el" href="classensembl_1_1variation_1_1VEPResult.html">ensembl.variation.VEPResult</a> Valid formats: json, xml HTTP endpoint: vep/:species/hgvs/:hgvs_notation</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>hgvs_notation (String) HGVS notation. May be genomic (g), coding (c) or protein (p), with reference to chromosome name, gene name, transcript ID or protein ID.</li>
</ul>
<p>Optional parameters:</p><ul>
<li>hgvs (Boolean) Include HGVS nomenclature based on Ensembl stable identifiers</li>
<li>ccds (Boolean) Include CCDS transcript identifiers</li>
<li>numbers (Boolean) Include affected exon and intron positions within the transcript</li>
<li>domains (Boolean) Include names of overlapping protein domains</li>
<li>canonical (Boolean) Include a flag indicating the canonical transcript for a gene</li>
<li>protein (Boolean) Include Ensembl protein identifiers</li>
<li>xref_refseq (Boolean) Include aligned RefSeq mRNA identifiers for transcript. NB: theRefSeq and Ensembl transcripts aligned in this way MAY NOT, AND FREQUENTLY WILL NOT, match exactly in sequence, exon structure and protein product </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l01158">1158</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-32" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-32-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-32-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-32-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a116a371b6785f9ff418c084a0bcd1fa4_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a116a371b6785f9ff418c084a0bcd1fa4_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a116a371b6785f9ff418c084a0bcd1fa4_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a116a371b6785f9ff418c084a0bcd1fa4_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="318,5,511,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="561,5,754,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aa11643f4c11ac605e58e1d8f2983486c"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getVariantConsequencesByRegionAllele </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>allele</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Fetch variant consequences. </p>
<p>Return type: <a class="el" href="classensembl_1_1variation_1_1VEPResult.html">ensembl.variation.VEPResult</a> Valid formats: json, xml HTTP endpoint: vep/:species/region/:region/:allele/</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>region (String) Query region. We only support the current assembly</li>
<li>allele (String) Variation allele</li>
</ul>
<p>Optional parameters:</p><ul>
<li>hgvs (Boolean) Include HGVS nomenclature based on Ensembl stable identifiers</li>
<li>ccds (Boolean) Include CCDS transcript identifiers</li>
<li>numbers (Boolean) Include affected exon and intron positions within the transcript</li>
<li>domains (Boolean) Include names of overlapping protein domains</li>
<li>canonical (Boolean) Include a flag indicating the canonical transcript for a gene</li>
<li>protein (Boolean) Include Ensembl protein identifiers</li>
<li>xref_refseq (Boolean) Include aligned RefSeq mRNA identifiers for transcript. NB: theRefSeq and Ensembl transcripts aligned in this way MAY NOT, AND FREQUENTLY WILL NOT, match exactly in sequence, exon structure and protein product </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l01092">1092</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-33" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-33-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-33-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-33-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_aa11643f4c11ac605e58e1d8f2983486c_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_aa11643f4c11ac605e58e1d8f2983486c_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_aa11643f4c11ac605e58e1d8f2983486c_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_aa11643f4c11ac605e58e1d8f2983486c_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="281,11,474,53"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="523,11,717,53"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac085e7a91b67964be9fb4487711f50bb"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getVariantConsequencesByVariationID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Fetch variant consequences based on a variation identifier. </p>
<p>Return type: <a class="el" href="classensembl_1_1variation_1_1VEPResult.html">ensembl.variation.VEPResult</a> Valid formats: json, xml HTTP endpoint: vep/:species/id/:id</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>id (String) Query ID. Supports dbSNP, COSMIC and HGMD identifiers</li>
</ul>
<p>Optional parameters:</p><ul>
<li>hgvs (Boolean) Include HGVS nomenclature based on Ensembl stable identifiers</li>
<li>ccds (Boolean) Include CCDS transcript identifiers</li>
<li>numbers (Boolean) Include affected exon and intron positions within the transcript</li>
<li>domains (Boolean) Include names of overlapping protein domains</li>
<li>canonical (Boolean) Include a flag indicating the canonical transcript for a gene</li>
<li>protein (Boolean) Include Ensembl protein identifiers</li>
<li>xref_refseq (Boolean) Include aligned RefSeq mRNA identifiers for transcript. NB: theRefSeq and Ensembl transcripts aligned in this way MAY NOT, AND FREQUENTLY WILL NOT, match exactly in sequence, exon structure and protein product </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l01125">1125</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-34" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-34-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-34-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-34-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_ac085e7a91b67964be9fb4487711f50bb_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_ac085e7a91b67964be9fb4487711f50bb_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_ac085e7a91b67964be9fb4487711f50bb_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_ac085e7a91b67964be9fb4487711f50bb_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="281,11,474,53"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="523,11,717,53"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ad9ef94b7e88cd5ddb3671efd0e538c08"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.getVariationByID </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Uses a variation identifier (e.g. </p>
<p>rsID) to return the variation features</p>
<p>Return type: <a class="el" href="classensembl_1_1variation_1_1Variation.html">ensembl.variation.Variation</a> Valid formats: json, xml HTTP endpoint: variation/:species/:id</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>id (String) Variation id</li>
</ul>
<p>Optional parameters:</p><ul>
<li>genotypes (Boolean(0,1)) Include individual genotypes</li>
<li>phenotypes (Boolean(0,1)) Include phenotypes</li>
<li>pops (Boolean(0,1)) Include population allele frequencies</li>
<li>population_genotypes (Boolean(0,1)) Include population genotype frequencies </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l01057">1057</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-35" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-35-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-35-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-35-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_ad9ef94b7e88cd5ddb3671efd0e538c08_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_ad9ef94b7e88cd5ddb3671efd0e538c08_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_ad9ef94b7e88cd5ddb3671efd0e538c08_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_ad9ef94b7e88cd5ddb3671efd0e538c08_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a34b21d974cb859e1bd43f09519a991f2"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.listAvailableReleases </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Shows the data releases available on this REST server. </p>
<p>May return more than one release (unfrequent non-standard Ensembl configuration).</p>
<p>Return type: None Valid formats: json, xml HTTP endpoint: info/data </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00459">459</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-36" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-36-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-36-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-36-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a34b21d974cb859e1bd43f09519a991f2_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a34b21d974cb859e1bd43f09519a991f2_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a34b21d974cb859e1bd43f09519a991f2_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a34b21d974cb859e1bd43f09519a991f2_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a29bcf93ab6da343c01898a3249bfc1c7"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.listComparaDatabases </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Lists all available comparative genomics databases and their data release. </p>
<p>Return type: None Valid formats: json, xml HTTP endpoint: info/comparas </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00420">420</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-37" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-37-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-37-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-37-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a29bcf93ab6da343c01898a3249bfc1c7_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a29bcf93ab6da343c01898a3249bfc1c7_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a29bcf93ab6da343c01898a3249bfc1c7_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a29bcf93ab6da343c01898a3249bfc1c7_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="251,5,445,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="494,5,687,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="acbdb1f715bc69a5d1247ad8657a935d6"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.listSpecies </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Lists all available species, their aliases, available adaptor groups and data release. </p>
<p>Return type: <a class="el" href="classensembl_1_1info_1_1Species.html">ensembl.info.Species</a> Valid formats: json, xml HTTP endpoint: info/species</p>
<p>Optional parameters:</p><ul>
<li>division (String) Filter by Ensembl or Ensembl Genomes division. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00407">407</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-38" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-38-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-38-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-38-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_acbdb1f715bc69a5d1247ad8657a935d6_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_acbdb1f715bc69a5d1247ad8657a935d6_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_acbdb1f715bc69a5d1247ad8657a935d6_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_acbdb1f715bc69a5d1247ad8657a935d6_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ab216ec4cd25019e26ed2c7f36dd653ab"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.lookupGeneSymbol </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>symbol</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find the species and database for a symbol in a linked external database. </p>
<p>Return type: <a class="el" href="namespaceensembl_1_1genome.html#a01899a318109093529836f503d161096" title="Wrapper arround the various types of features. ">ensembl.genome.feature_wrapper</a> Valid formats: json, xml HTTP endpoint: lookup/symbol/:species/:symbol</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>symbol (String) A name or symbol from an annotation source has been linked to a genetic feature</li>
</ul>
<p>Optional parameters:</p><ul>
<li>format (Enum(full,condensed)) Specify the layout of the response</li>
<li>expand (Boolean(0,1)) Expands the search to include any connected features. e.g. If the object is a gene, its transcripts, translations and exons will be returned as well. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00595">595</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-39" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-39-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-39-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-39-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_ab216ec4cd25019e26ed2c7f36dd653ab_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_ab216ec4cd25019e26ed2c7f36dd653ab_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_ab216ec4cd25019e26ed2c7f36dd653ab_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_ab216ec4cd25019e26ed2c7f36dd653ab_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a49d1054ef7f54d12f6e2689d0f41275d"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.lookupIdentifier </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Find the species and database for a single identifier. </p>
<p>Return type: <a class="el" href="namespaceensembl_1_1genome.html#a01899a318109093529836f503d161096" title="Wrapper arround the various types of features. ">ensembl.genome.feature_wrapper</a> Valid formats: json, xml HTTP endpoint: lookup/id/:id</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>db_type (String) Restrict the search to a database other than the default. Useful if you need to use a DB other than core</li>
<li>format (Enum(full,condensed)) Specify the formats to emit from this endpoint</li>
<li>expand (Boolean(0,1)) Expands the search to include any connected features. e.g. If the object is a gene, its transcripts, translations and exons will be returned as well. </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00572">572</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-40" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-40-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-40-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-40-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a49d1054ef7f54d12f6e2689d0f41275d_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a49d1054ef7f54d12f6e2689d0f41275d_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a49d1054ef7f54d12f6e2689d0f41275d_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a49d1054ef7f54d12f6e2689d0f41275d_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="ac03cfdf0f4311e40cc90b74ae235374e"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.mapCDNACoordinatesToGenome </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Convert from cDNA coordinates to genomic coordinates. </p>
<p>Output reflects forward orientation coordinates as returned from the Ensembl API.</p>
<p>Return type: <a class="el" href="classensembl_1_1genome_1_1Location.html">ensembl.genome.Location</a> Valid formats: json, xml HTTP endpoint: map/cdna/:id/:region</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
<li>region (String) Query region</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00641">641</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-41" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-41-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-41-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-41-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_ac03cfdf0f4311e40cc90b74ae235374e_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_ac03cfdf0f4311e40cc90b74ae235374e_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_ac03cfdf0f4311e40cc90b74ae235374e_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_ac03cfdf0f4311e40cc90b74ae235374e_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="318,5,511,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="561,5,754,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aec0a48bef0d6b6382c5ea9e05bc2fe71"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.mapCDSCoordinatesToGenome </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Convert from CDS coordinates to genomic coordinates. </p>
<p>Output reflects forward orientation coordinates as returned from the Ensembl API.</p>
<p>Return type: <a class="el" href="classensembl_1_1genome_1_1Location.html">ensembl.genome.Location</a> Valid formats: json, xml HTTP endpoint: map/cds/:id/:region</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
<li>region (String) Query region</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00662">662</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-42" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-42-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-42-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-42-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_aec0a48bef0d6b6382c5ea9e05bc2fe71_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_aec0a48bef0d6b6382c5ea9e05bc2fe71_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_aec0a48bef0d6b6382c5ea9e05bc2fe71_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_aec0a48bef0d6b6382c5ea9e05bc2fe71_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="307,5,501,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="550,5,743,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="aff6b6b851a8290ccd31e9d4ad8faae90"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.mapCoordinatesBetweenAssemblies </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>species</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>asm_one</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>asm_two</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Convert the co-ordinates of one assembly to another. </p>
<p>Return type: <a class="el" href="classensembl_1_1genome_1_1CoordMapping.html">ensembl.genome.CoordMapping</a> Valid formats: json, xml HTTP endpoint: map/:species/:asm_one/:region/:asm_two</p>
<p>Required parameters:</p><ul>
<li>species (String) Species name/alias</li>
<li>asm_one (String) Version of the input assembly</li>
<li>region (String) Query region</li>
<li>asm_two (String) Version of the output assembly</li>
</ul>
<p>Optional parameters:</p><ul>
<li>coord_system (String) Name of the output coordinate system </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00620">620</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-43" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-43-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-43-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-43-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_aff6b6b851a8290ccd31e9d4ad8faae90_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_aff6b6b851a8290ccd31e9d4ad8faae90_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_aff6b6b851a8290ccd31e9d4ad8faae90_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_aff6b6b851a8290ccd31e9d4ad8faae90_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="334,5,527,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="577,5,770,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a2efad4a28b137e36fbbe3ca11ac211be"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.mapProteinCoordinatesToGenome </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>id</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>region</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Convert from protein (translation) coordinates to genomic coordinates. </p>
<p>Output reflects forward orientation coordinates as returned from the Ensembl API.</p>
<p>Return type: <a class="el" href="classensembl_1_1genome_1_1Location.html">ensembl.genome.Location</a> Valid formats: json, xml HTTP endpoint: map/translation/:id/:region</p>
<p>Required parameters:</p><ul>
<li>id (String) An Ensembl stable ID</li>
<li>region (String) Query region</li>
</ul>
<p>Optional parameters:</p><ul>
<li>species (String) Species name/alias </li>
</ul>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00683">683</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-44" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-44-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-44-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-44-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a2efad4a28b137e36fbbe3ca11ac211be_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a2efad4a28b137e36fbbe3ca11ac211be_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a2efad4a28b137e36fbbe3ca11ac211be_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a2efad4a28b137e36fbbe3ca11ac211be_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="318,5,511,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="561,5,754,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a1e5af2358db2f18df52de359690ad446"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.ping </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Checks if the service is alive. </p>
<p>Return type: None Valid formats: json, xml HTTP endpoint: info/ping </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00391">391</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-45" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-45-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-45-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-45-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a1e5af2358db2f18df52de359690ad446_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a1e5af2358db2f18df52de359690ad446_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a1e5af2358db2f18df52de359690ad446_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a1e5af2358db2f18df52de359690ad446_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<a class="anchor" id="a28582fe2c741d95c84538de986cce616"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def ensembl._pyrest_server.RestServer.rest_version </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>self</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>kwargs</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Shows the current version of the Ensembl REST API. </p>
<p>Return type: None Valid formats: json, xml HTTP endpoint: info/rest </p>

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00446">446</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

<p><div id="dynsection-46" onclick="return toggleVisibility(this)" class="dynheader closed" style="cursor:pointer;">
  <img id="dynsection-46-trigger" src="closed.png" alt="+"/> Here is the call graph for this function:</div>
<div id="dynsection-46-summary" class="dynsummary" style="display:block;">
</div>
<div id="dynsection-46-content" class="dyncontent" style="display:none;">
<div class="center"><img src="classensembl_1_1__pyrest__server_1_1RestServer_a28582fe2c741d95c84538de986cce616_cgraph.png" border="0" usemap="#classensembl_1_1__pyrest__server_1_1RestServer_a28582fe2c741d95c84538de986cce616_cgraph" alt=""/></div>
<map name="classensembl_1_1__pyrest__server_1_1RestServer_a28582fe2c741d95c84538de986cce616_cgraph" id="classensembl_1_1__pyrest__server_1_1RestServer_a28582fe2c741d95c84538de986cce616_cgraph">
<area shape="rect" id="node2" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a071038622ab8bb772af3e343cc8288af" title="ensembl._pyrest_server.Rest\lServer.__build_rest_answer" alt="" coords="249,5,442,46"/><area shape="rect" id="node3" href="classensembl_1_1__pyrest__server_1_1RestServer.html#a911a9bd143963eea035e9188abd86e77" title="ensembl._pyrest_server.Rest\lServer.__get_json_answer" alt="" coords="491,5,685,46"/></map>
</div>
</p>

</div>
</div>
<h2 class="groupheader">Member Data Documentation</h2>
<a class="anchor" id="a36aea40c33230383043e8291e5a9022e"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">dictionary ensembl._pyrest_server.RestServer.__content_types</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<b>Initial value:</b><div class="fragment"><div class="line"><a name="l00001"></a><span class="lineno">    1</span>&#160;= {</div>
<div class="line"><a name="l00002"></a><span class="lineno">    2</span>&#160;    <span class="stringliteral">&quot;bed&quot;</span>: <span class="stringliteral">&quot;text/x-bed&quot;</span>,</div>
<div class="line"><a name="l00003"></a><span class="lineno">    3</span>&#160;<span class="stringliteral">&quot;fasta&quot;</span>: <span class="stringliteral">&quot;text/x-fasta&quot;</span>,</div>
<div class="line"><a name="l00004"></a><span class="lineno">    4</span>&#160;<span class="stringliteral">&quot;gff3&quot;</span>: <span class="stringliteral">&quot;text/x-gff3&quot;</span>,</div>
<div class="line"><a name="l00005"></a><span class="lineno">    5</span>&#160;<span class="stringliteral">&quot;json&quot;</span>: <span class="stringliteral">&quot;application/json&quot;</span>,</div>
<div class="line"><a name="l00006"></a><span class="lineno">    6</span>&#160;<span class="stringliteral">&quot;nh&quot;</span>: <span class="stringliteral">&quot;text/x-nh&quot;</span>,</div>
<div class="line"><a name="l00007"></a><span class="lineno">    7</span>&#160;<span class="stringliteral">&quot;phyloxml&quot;</span>: <span class="stringliteral">&quot;text/x-phyloxml+xml&quot;</span>,</div>
<div class="line"><a name="l00008"></a><span class="lineno">    8</span>&#160;<span class="stringliteral">&quot;xml&quot;</span>: <span class="stringliteral">&quot;text/xml&quot;</span>,</div>
<div class="line"><a name="l00009"></a><span class="lineno">    9</span>&#160;<span class="stringliteral">&quot;yaml&quot;</span>: <span class="stringliteral">&quot;text/x-yaml&quot;</span>,</div>
<div class="line"><a name="l00010"></a><span class="lineno">   10</span>&#160;<span class="stringliteral">&quot;text&quot;</span>: <span class="stringliteral">&quot;text/plain&quot;</span></div>
<div class="line"><a name="l00011"></a><span class="lineno">   11</span>&#160;    }</div>
</div><!-- fragment -->
<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00041">41</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

</div>
</div>
<a class="anchor" id="adfef2141465bd19e31b86d7ebafc3524"></a>
<div class="memitem">
<div class="memproto">
<table class="mlabels">
  <tr>
  <td class="mlabels-left">
      <table class="memname">
        <tr>
          <td class="memname">dictionary ensembl._pyrest_server.RestServer.__return_codes</td>
        </tr>
      </table>
  </td>
  <td class="mlabels-right">
<span class="mlabels"><span class="mlabel">static</span><span class="mlabel">private</span></span>  </td>
  </tr>
</table>
</div><div class="memdoc">
<b>Initial value:</b><div class="fragment"><div class="line"><a name="l00001"></a><span class="lineno">    1</span>&#160;= {</div>
<div class="line"><a name="l00002"></a><span class="lineno">    2</span>&#160;    200: (<span class="stringliteral">&quot;OK&quot;</span>, <span class="stringliteral">&quot;Request was a success&quot;</span>),</div>
<div class="line"><a name="l00003"></a><span class="lineno">    3</span>&#160;400: (<span class="stringliteral">&quot;Bad Request&quot;</span>, <span class="stringliteral">&quot;Occurs during exceptional circumstances such as the service is unable to find an ID&quot;</span>),</div>
<div class="line"><a name="l00004"></a><span class="lineno">    4</span>&#160;404: (<span class="stringliteral">&quot;Not Found&quot;</span>, <span class="stringliteral">&quot;Indicates a badly formatted request. Check your URL&quot;</span>),</div>
<div class="line"><a name="l00005"></a><span class="lineno">    5</span>&#160;429: (<span class="stringliteral">&quot;Too Many Requests&quot;</span>, <span class="stringliteral">&quot;You have been rate-limited; wait and retry&quot;</span>),</div>
<div class="line"><a name="l00006"></a><span class="lineno">    6</span>&#160;503: (<span class="stringliteral">&quot;Service Unavailable&quot;</span>, <span class="stringliteral">&quot;The service is temporarily down; retry after a pause&quot;</span>)</div>
<div class="line"><a name="l00007"></a><span class="lineno">    7</span>&#160;    }</div>
</div><!-- fragment -->
<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00053">53</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

</div>
</div>
<a class="anchor" id="a14663d4e51e10889625b7a917a9b5347"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">ensembl._pyrest_server.RestServer.http</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00068">68</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

</div>
</div>
<a class="anchor" id="a41af4531182edc3f2597d8c7ecf2c33f"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">ensembl._pyrest_server.RestServer.last_headers</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00069">69</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

</div>
</div>
<a class="anchor" id="a2bd4a8528904fd5e23ce4ddec3c90652"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">ensembl._pyrest_server.RestServer.server_url</td>
        </tr>
      </table>
</div><div class="memdoc">

<p>Definition at line <a class="el" href="__pyrest__server_8py_source.html#l00067">67</a> of file <a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a>.</p>

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li>ensembl/<a class="el" href="__pyrest__server_8py_source.html">_pyrest_server.py</a></li>
</ul>
</div><!-- contents -->
</div><!-- doc-content -->
<!-- start footer part -->
<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
  <ul>
    <li class="navelem"><a class="el" href="namespaceensembl.html">ensembl</a></li><li class="navelem"><a class="el" href="namespaceensembl_1_1__pyrest__server.html">_pyrest_server</a></li><li class="navelem"><a class="el" href="classensembl_1_1__pyrest__server_1_1RestServer.html">RestServer</a></li>
    <li class="footer">Generated by
    <a href="http://www.doxygen.org/index.html">
    <img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.8.7 </li>
  </ul>
</div>
</body>
</html>
